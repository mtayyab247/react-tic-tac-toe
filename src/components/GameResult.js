import React from 'react';

const GameResult = () => {

    return(
        <div className="winner-wrapper">
        </div>
    );
};

export default GameResult;