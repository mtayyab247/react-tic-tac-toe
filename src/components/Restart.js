import React, {useState, useEffect} from 'react';

const Restart = ({selectionCount, handleFinish}) => {

    const[restartBtnVisibility, setRrestartBtnVisibility] = useState("hide");
    useEffect(() => {
        if(selectionCount > 1) {
            setRrestartBtnVisibility("show");
        }
    }, [selectionCount]);

    const handleRestartBtn = () => {
        handleFinish();
        setRrestartBtnVisibility("hide");
    };

    return(
        <div className="restart-wrapper">
            <button onClick={handleRestartBtn} className={restartBtnVisibility + " restart"}>Restart</button>
        </div>
    );
};

export default Restart;