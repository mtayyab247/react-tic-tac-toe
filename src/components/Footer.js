import React from 'react';
import Restart from './Restart';
import GameResult from './GameResult';
import '../css/footer.css';

const Footer = ({winner, playerTurn, handleFinish, selectionCount}) => {
    return(
        <div className="footer">
            <Restart handleFinish={handleFinish} selectionCount={selectionCount} />
            <GameResult winner={winner} playerTurn={playerTurn} />
        </div>
    );
};

export default Footer;