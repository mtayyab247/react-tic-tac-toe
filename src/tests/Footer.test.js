import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Footer from '../components/Footer';
import Restart from '../components/Restart';
import GameResult from '../components/GameResult';

Enzyme.configure({
    adapter: new Adapter()
});

describe("<Footer /> test", () => {
    it("Should render Restart and Result", () => {
        const wrapper = shallow(<Footer />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("Should render Restart", () => {
        const wrapper = shallow(<Restart />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("Should render Result", () => {
        const wrapper = shallow(<GameResult />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});


