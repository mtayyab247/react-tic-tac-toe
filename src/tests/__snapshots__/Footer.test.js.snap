// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> test Should render Restart 1`] = `
<div
  className="restart-wrapper"
/>
`;

exports[`<Footer /> test Should render Restart and Result 1`] = `
<div
  className="footer"
>
  <Restart />
</div>
`;

exports[`<Footer /> test Should render Result 1`] = `
<div
  className="winner-wrapper"
/>
`;
